src/gridfs-stream/index.ts

```diff
@@ -1,53 +1,65 @@
 import { EventEmitter } from 'events';
-import { GridFSBucketReadStream } from './download';
-import { GridFSBucketWriteStream } from './upload';
+import { GridFSBucketReadStream, GridFSBucketReadStreamOptions } from './download';
+import { GridFSBucketWriteStream, GridFSBucketWriteStreamOptions, Id } from './upload';
 import { executeLegacyOperation } from '../utils';
-import type { Callback } from '../types';
+import { WriteConcernOptions, WriteConcern } from '../write_concern';
+import type { Callback, AnyError } from '../types';
+import type { Db, ReadPreference } from '..';
+import type { Collection } from '../collection';
+import type { Document } from './../types.d';
+import type { Cursor } from './../cursor/cursor';
+import type { FindOptions } from './../operations/find';
 import { MongoError } from '../error';
+import type { Logger } from '../logger';
 
-const DEFAULT_GRIDFS_BUCKET_OPTIONS: any = {
+const DEFAULT_GRIDFS_BUCKET_OPTIONS = {
   bucketName: 'fs',
   chunkSizeBytes: 255 * 1024
 };
 
+interface GridFSBucketOptions extends WriteConcernOptions {
+  /** The 'files' and 'chunks' collections will be prefixed with the bucket name followed by a dot. */
+  bucketName?: string;
+  /** Number of bytes stored in each chunk. Defaults to 255KB */
+  chunkSizeBytes?: number;
+  /** Read preference to be passed to read operations */
+  readPreference?: ReadPreference;
+}
+
+interface GridFSBucketPrivate {
+  db: Db;
+  options: {
```

Comment:
This was deliberate. The internal options are non-optional because we apply defaults. Ideally properties like `bucketName` would be `this.bucketName`, same for `writeConcern` is also internally narrower, again can be `this.writeConcern`.  Let me know what your preference is here.